### PR TITLE
Remove extension when folder is removed during runtime

### LINKED ISSUE #1461: Lens should notice added/removed extension folders
Currently Lens (ExtensionManager) scans extension folder only during boot/init. We should improve this so that Lens detects added/removed folders and notifies ExtensionLoader.

Relates to #1456 and #1199

### PR DESCRIPTION
**Pull request by panuhorsmalahti**
- prettier is added only to support inline snapshots
- If user has extension page open and the extension folder is deleted, "Page not found" 'error' will be displayed. We could redirect to a working page. I'll add issue for this.

Fixes #1461

### CODE REVIEW

File: src/extensions/extension-loader.ts [hunk 1]
```diff
@@ -57,11 +64,28 @@ export class ExtensionLoader {
...
+    if (instance) {
+      try {
+        instance.disable();
+        this.events.emit("remove", instance);
+        this.instances.delete(lensExtensionId);
+      } catch (error) {
+        logger.error(`${logModule}: deactivation extension error`, { lensExtensionId, error });
+      }
+    }
+  }
```
Comment (Nokel81) [inline src/extensions/extension-loader.ts:80]: What do you think about the `if (!instance) { return; }` style instead?

File: src/extensions/extension-loader.ts [hunk 2]
```diff
@@ -57,11 +64,28 @@ export class ExtensionLoader {
...
+
   removeExtension(lensExtensionId: LensExtensionId) {
-    // TODO: Remove the extension properly (from menus etc.)
+    this.removeInstance(lensExtensionId);
+
     if (!this.extensions.delete(lensExtensionId)) {
       throw new Error(`Can't remove extension ${lensExtensionId}, doesn't exist.`);
     }
+
```
Comment (Nokel81) [inline src/extensions/extension-loader.ts:88]: New line at the end of a function?

File: yarn.lock [hunk 3]
```diff
@@ -11480,6 +11480,11 @@ prepend-http@^2.0.0:
   resolved "https://registry.yarnpkg.com/prepend-http/-/prepend-http-2.0.0.tgz#e92434bfa5ea8c19f41cdfd401d741a3c819d897"
   integrity sha1-6SQ0v6XqjBn0HN/UAddBo8gZ2Jc=
 
+prettier@^2.2.0:
+  version "2.2.0"
+  resolved "https://registry.yarnpkg.com/prettier/-/prettier-2.2.0.tgz#8a03c7777883b29b37fb2c4348c66a78e980418b"
+  integrity sha512-yYerpkvseM4iKD/BXLYUkQV5aKt4tQPqaGW6EsZjzyu0r7sVZZNPJW4Y8MyKmicp6t42XUPcBVA+H6sB3gqndw==
+
```
Comment (Nokel81) [inline yarn.lock:11487]: I don't see any references to it in any of this diff. Are you sure you are actually using it for your intended purpose?

Comment (panuhorsmalahti) [reply yarn.lock:11487]: That was mentioned in the issue description, jest requires this dependency for inline snapshots: https://github.com/facebook/jest/issues/8467

Comment (Nokel81) [reply yarn.lock:11487]: Ah I saw that comment, but I assumed that it would have to be explicitly "used" somewhere.

**Review by jakolehm — APPROVED** [review]

File: src/extensions/extension-loader.ts (hunk 1 above)
Comment (panuhorsmalahti) [reply src/extensions/extension-loader.ts:80]: I don't see that being necessarily better, but fixed

File: src/extensions/extension-loader.ts (hunk 2 above)
Comment (panuhorsmalahti) [reply src/extensions/extension-loader.ts:88]: Fixed

**Pull request merged**